File: app/src/main/java/com/mobile/rxjava2andretrofit2/login/model/LoginModelImpl.java
package com.mobile.rxjava2andretrofit2.login.model;

import com.mobile.rxjava2andretrofit2.login.request.LoginRequest;
import com.mobile.rxjava2andretrofit2.manager.RetrofitManager;

import java.util.Map;

import io.reactivex.Observable;
import okhttp3.ResponseBody;

/**
 * author    : xxxxxxxxxxx
 * e-mail    : xxxxxxxxxxx@qq.com
 * date      : 2019/3/10 17:03
 * introduce :
 */

public class LoginModelImpl implements ILoginModel {

    private static final String TAG = "LoginModelImpl";

    public LoginModelImpl() {
    }

    @Override
    public Observable<ResponseBody> login(Map<String, String> bodyParams) {
        return RetrofitManager.getInstance().getRetrofit()
                .create(LoginRequest.class)
                .getLoginData(bodyParams);
    }

    @Override
    public Observable<ResponseBody> register(Map<String, String> bodyParams) {
        return RetrofitManager.getInstance().getRetrofit()
                .create(LoginRequest.class)
                .getRegisterData(bodyParams);
    }

}

File: library_login/src/main/java/com/phone/library_login/model/LoginModelImpl.kt
package com.phone.library_login.model

import com.phone.library_network.manager.RetrofitManager
import com.phone.library_login.request.LoginRequest
import io.reactivex.Observable
import okhttp3.ResponseBody

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      : 2019/3/10 17:03
 * introduce :
 */

class LoginModelImpl : ILoginModel {

    private val TAG = LoginModelImpl::class.java.simpleName

    override fun getAuthCode(bodyParams: Map<String, String>): Observable<ResponseBody> {
        return RetrofitManager.instance.mRetrofit
            .create(LoginRequest::class.java)
            .getAuthCode(bodyParams)
    }

    override fun loginWithAuthCode(bodyParams: Map<String, String>): Observable<ResponseBody> {
        return RetrofitManager.instance.mRetrofit
            .create(LoginRequest::class.java)
            .getLoginWithAuthCodeData(bodyParams)
    }

    override fun register(bodyParams: Map<String, String>): Observable<ResponseBody> {
        return RetrofitManager.instance.mRetrofit
            .create(LoginRequest::class.java)
            .getRegisterData(bodyParams)
    }

}